src/coprocessor/dag/rpn_expr/types/expr_eval.rs

```diff
@@ -243,32 +241,37 @@ impl RpnExpression {
                     });
                 }
                 RpnExpressionNode::FnCall {
-                    ref func,
-                    ref field_type,
-                    ref implicit_args,
+                    func_meta,
+                    args_len,
+                    field_type: ret_field_type,
+                    implicit_args,
                 } => {
                     // Suppose that we have function call `Foo(A, B, C)`, the RPN nodes looks like
                     // `[A, B, C, Foo]`.
                     // Now we receives a function call `Foo`, so there are `[A, B, C]` in the stack
                     // as the last several elements. We will directly use the last N (N = number of
                     // arguments) elements in the stack as function arguments.
-                    assert!(stack.len() >= func.args_len);
-                    let stack_slice_begin = stack.len() - func.args_len;
+                    assert!(stack.len() >= *args_len);
+                    let stack_slice_begin = stack.len() - *args_len;
                     let stack_slice = &stack[stack_slice_begin..];
-                    let call_info = RpnFnCallPayload {
-                        output_rows,
-                        raw_args: stack_slice,
+                    let mut call_extra = RpnFnCallExtra {
+                        ret_field_type,
                         implicit_args,
-                        ret_field_type: field_type,
                     };
-                    let ret = (func.fn_ptr)(context, &call_info)?;
+                    let ret = (func_meta.fn_ptr)(
```

Comment:
These arguments are passed fairly deep in stacks. So it can be expensive to copy these arguments between stack frames if the compiler is not smart enough. Perhaps we'd better still use a payload struct to wrap them all.

Reply:
I changed it because according to AMD64 ABI these parameters should be kept in the registers when the function is invoked. Also if there is demand to store these parameters onto stack, a single `pushq` should be enough for each parameter. Notice that this may behave differently compared to the `RpnFnCallPayload` previously since `RpnFnCallPayload` is a integral structure while compiler is very likely to generate simple `mov` instructions for these scattered parameters.